### DISCUSSION TITLE
String interpolation, maybe?

### DISCUSSION THREAD

**Discussion by vascanera** [odin-lang/Odin#1010] [Ideas/Requests]
I know that there was some closed thread (and rejected proposal) in "Issues" with some proposals, that, among other stuff, had some proposal for string interpolation (https://github.com/odin-lang/Odin/issues/169). 

Could this feature (string interpolation), eventually, make it into the language, in one form or another? 

In latest available Tag (dev-2021-06) you added "Unified cond ? x : y and x if cond else y logic" which, actually, gives us the same result using 2 different syntax constructs.
A string interpolation implementation could be treated in the same way, as a re-arranged syntax for printf. 

Will we ever get this into the language? 
I think the usefulness of such a feature is quite obvious when you need to print lots of data interpolated with other text.
Is this way out of line with Odin's philosophy?

Cheers! Keep up the good work.

**Comment by ThisDevDane** [comment]
I think one problem Bill has with it, is that it is sort of a "hidden" memory allocation and a hidden import (`import "core:fmt"`), personally I would also like string interpolation in Odin but I don't really find it too annoying just having to import fmt and just do the full statement.

**Reply by vascanera** [reply]
Yes, it makes perfect sense. fmt package must be imported. Totally forgot about that.

**Reply by vascanera** [reply]
P.S.: according to what Bill said in "Conventionality and Idiomatic Constructs" (Youtube), "Sugar optimizes for typing, NOT for reading.". But, the sugar invloved in string interpolation brings more horizontal (left-to-right) linear readability and typeability, and it's easier to reason about compared to, say, printf, on which you have to the the mental gymnastics of counting and mapping the replaced  parameters with their placeholders inside the string. Just my 2cents here.

**Reply by gingerBill** [reply]
I actually find string interpolation harder to read most of the time than the standard approach, and it's also worse when you have to deal with things like locali[sz]ation.

I will never support string interpolation on the ground that it requires implicit and hidden memory allocations alone. Use `package fmt` instead.
